src/binary/raw_binary_reader.rs

```diff
@@ -947,12 +975,22 @@ where
         use IonTypeCode::*;
         let length = match header.ion_type_code {
             NullOrNop | Boolean => 0,
-            PositiveInteger | NegativeInteger | Decimal | Timestamp | String | Symbol | List
-            | SExpression | Clob | Blob => self.read_standard_length()?,
+            PositiveInteger | NegativeInteger | Decimal | String | Symbol | List | SExpression | Clob | Blob => {
+                self.read_standard_length()?
+            }
+            Timestamp => {
+                let length = self.read_standard_length()?;
+                if length <= 1 && !self.cursor.value.is_null {
+                    return decoding_error(
+                        "found a non-null timestamp (typecode=6) with a length <= 1",
```

Comment:
🗺️ Binary `timestamp`s [must specify a `year` and `offset`](https://amzn.github.io/ion-docs/docs/binary.html#0x6-timestamp), making their minimum legal length `2`.